### ISSUE TITLE
Deployed Web App Offers Your Own Data Only Responses

### ISSUE THREAD

**Issue by raeesgillani** [microsoft/sample-app-aoai-chatGPT#278]
Hi,

when testing with the model using my own data inside Azure AI Studio, I am able to ask it questions like 'who is microsoft', alongside questions about my data and it is able to give good answers. However, when asking the same question via the deployed Web App, it responds with 'cannot find data' on questions like 'who is microsoft' but is able to answer questions about my data fine.

Is it this a bug, or a configuration issues? - please advise, how can it be resolved?

**Comment by peterjeitschko** [comment]
I have the exact same problem. It seems to me that none of the parameters get's passed to the webapp. Even if you do to the configuration of the app, parameters like top K search results do not take into effect.

**Comment by peterjeitschko** [comment]
@jonathanpv allthough the inScope value is set to "false", the setting seem to have an effect when  deploying the Webapp. The playground works fine

**Comment by davidm00** [comment]
Hi @raeesgillani, @peterjeitschko  is the response you are receiving something like this: "The requested information is not found in the retrieved data. Please try another query or topic."? If so, then I believe you may have checked the option to limit responses to your data only. Another thing, go to the azure portal and verify that your web app's environment variables are populated with the configurations that you selected in the Azure OpenAI studio. 

![image](https://github.com/microsoft/sample-app-aoai-chatGPT/assets/43946287/c4bd2c5b-4986-4678-a9d2-edd55f9d8469)

**Comment by raeesgillani** [comment]
After doing some reading, resolved the issue, so marking this as closed. @peterjeitschko is correct, when clicking the deploy to web app button inside Azure OpenAI Studio. Regardless of the configurations selected, the web app defaults as defined in this repo are used instead.

The workaround to get this working in the meantime, while I hope the Azure OpenAI team resolve the issue, is to use the manual deployment method indicated in the readme section of this repository. The manual method allows us to explicitly provide the various parameters which do get correctly deployed.

The only issues via the manual method I found were, I did not know the format of the embedding endpoint, could not find it as per the instructions, instead it is of the format: `[open_ai_resource_endpoint]/openai/deployments/[name_of_deployed_text_model]/embeddings?api-version=2023-07-01-preview`

Additionally, one of the downsides of using the deploy to azure button, is if you want to make frequent changes, it's better to clone the repo locally, test locally, and deploy after it works. As clicking the deploy to azure button multiple times ends up reaching the app service plan's subscription throttling limit.
